.github/workflows/security-trivy-scan-callable.yml

```diff
@@ -0,0 +1,123 @@
+name: Security - Scan Docker Image With Trivy
+
+on:
+  workflow_dispatch:
+    inputs:
+      image_ref:
+        description: 'Full image reference to scan e.g ghcr.io/n8n-io/n8n:latest'
+        required: true
+        default: 'ghcr.io/n8n-io/n8n:latest'
+  workflow_call:
+    inputs:
+      image_ref:
+        type: string
+        description: 'Full image reference to scan e.g ghcr.io/n8n-io/n8n:latest'
+        required: true
+    secrets:
+      SLACK_WEBHOOK_URL:
+        required: true
+
+jobs:
+  security_scan:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Pull Docker image
+        run: |
+          docker pull ${{ inputs.image_ref }}
+
+      - name: Run Trivy vulnerability scanner
```

Comment:
Shouldn't this step depend on pull success? Or if there's e.g. network issues then `docker pull whatever` already exits with `1`?

Reply:
If the previous step fails the rest of the workflow won't run. It would only be if we had "always()" as a condition or something similar it would run.